### ISSUE TITLE
text for https://carpentries-incubator.github.io/cwl-novice-tutorial/02-shell_to_cwl/index.html doesn't allign with exercises

### ISSUE THREAD

**Issue by swzCuroverse** [carpentries-incubator/cwl-novice-tutorial#54]
It seems like we need to address a bunch of topics in the shell to cwl text that isn't explained to do the exercises
1) output is first explained using stdout which I think is super confusing as a first example, additionally no examples are given to show how to do output that isn't stdout or a File but the example assume you know what to do that.
2) there is no example on how you use a glob that doesn't exactly define the file, you need to know what to do the examples (you could change it to be a *
3) no examples of basecommand with two inputs , etc -- 
baseCommand: [bwa, mem] and why you would do that instead of inputs
4) valueFrom is not shown but used in the exercise
5) InitialWorkDirRequirement is not described but used in the exercises

Personally, I think the echo example to be too much a toy example and doesn't really show the complexity of CWL and limits what you need to show i.e. different output types, valueFrom, glob, InitialWorkDirRequirement, etc.  Since you introduce the RNA seq example in the first part, it probably is a better example to showcase some of these ideas.

**Comment by ALuesink** [comment]
Hi @swzCuroverse, 
thank you for your remarks. This episode is still a work in progress and a lot will be changed, for example the exercises. The RNA-seq example will be the main part of this episode and will be used in later episodes.

**Comment by mr-c** [comment]
Thanks @swzCuroverse 

Today I realized that this novice tutorial is going to be more about using, combining, and adapting existing CommandLineTool descriptions into Workflows. Less so about writing CommandLineTool description's from scratch

**Comment by swzCuroverse** [comment]
How best should I help with this?  I can work to swap out the example from from the RNA-seq example.

**Comment by swzCuroverse** [comment]
"Today I realized that this novice tutorial is going to be more about using, combining, and adapting existing CommandLineTool descriptions into Workflows. Less so about writing CommandLineTool description's from scratch" <--- so we are assuming they have some commandline tools done already?  That seems like a strange subsection to focus on
 for intro content.

**Comment by douglowe** [comment]
IIRC, my intention for the first lesson was to use the writing of a simple CommandLineTool as an introduction to the basic CWL syntax (in a manner which allows for the learner to follow the transition from shell style syntax through to CWL style syntax). The end of the lesson was then to include the tool into a 1 step workflow, and the next lesson would be the move from 1-step to multi-step workflows.

To keep things moving in the later lessons we probably should provide the learners with the tool descriptions they're going to use. IIRC there will then be later lessons (in the intermediate CWL tutorial) which will cover some of the grittier details of CommandLineTool writing.

**Comment by mr-c** [comment]
> so we are assuming they have some commandline tools done already? That seems like a strange subsection to focus on for intro content.

Nope. Just as a Galaxy user almost never knows how to write Galaxy Tool XML, the beginning CWL user can learn workflow thinking without writing their own CommandLineTool descriptions

**Comment by mr-c** [comment]
> To keep things moving in the later lessons we probably should provide the learners with the tool descriptions they're going to use. IIRC there will then be later lessons (in the intermediate CWL tutorial) which will cover some of the grittier details of CommandLineTool writing.

Agreed!
